pytorch_lightning/overrides/torch_distributed.py

```diff
@@ -0,0 +1,175 @@
+# type: ignore
+
+import io
+import logging
+import os
+import pickle
+
+import torch
+from torch._C._distributed_c10d import ProcessGroup
+
+from pytorch_lightning.utilities import _HPU_AVAILABLE
+from pytorch_lightning.utilities.imports import _TORCH_GREATER_EQUAL_1_8
+
+_pickler = pickle.Pickler
+_unpickler = pickle.Unpickler
+
+
+logger = logging.getLogger(__name__)
+
+if torch.distributed.is_available():
+    from torch.distributed import Backend, broadcast, get_backend, get_rank, GroupMember
+
+# The code underneath is taken from PyTorch `torch/distributed/distributed_c10d.py`
+# the distributed backend and tensor type updates for habana backend is done here before broadcast
+
+
+# https://github.com/pytorch/pytorch/blob/master/torch/distributed/distributed_c10d.py#L256
```

Comment:
I appreciate sharing the line numbers and file link here - but since the file can be actively developed, I would prefer to either:

1. Share the permanent link, so that we don't have to update this link later.
2. Share just the file link, and users can search for the function there.

(please consider this suggestion for all instances in this file wherever the link to PyTorch code is shared)

What do you all think?

Reply:
Updated with links to a specific commit. Thanks! :)